### ISSUE TITLE
[16.0] On payment order Sepa Direct Debit, partner bank is not rirg

### ISSUE THREAD

**Issue by angelmoya** [OCA/bank-payment#1193]
Same issue #1131 
reported by @cesarsc79 @10537

## Module
accoun_banking_sepa_direct_debit, but really the error is on account_payment_order module.

## Describe the bug
I can not replicate on runbot, but I'm not alone.

I create a payment order (sepa direct debit):
- invoice is ok
- customer is ok
- bank number is sepa and related with right bank, ok
- mandate is ok

When payment order is confirmed, the account.payment generated take the company account, not the partner account.

So the file generated is not right, because the company bank account is set DbtrAcct

## What's going on?

The function _prepare_account_payment_vals() is rigth, debuggin I can check that is the partner bank account

https://github.com/OCA/bank-payment/blob/55718af9a49b00598d06db2d6a045deb9cddd530/account_payment_order/models/account_payment_line.py#L184

But, partner_bank_account is computed stored:

```
    partner_bank_id = fields.Many2one('res.partner.bank', string="Recipient Bank Account",
        readonly=False, store=True, tracking=True,
        compute='_compute_partner_bank_id',
        domain="[('id', 'in', available_partner_bank_ids)]",
        check_company=True)
```

As sepa direct debit is incoming payment, the domain is set to company bank account.

https://github.com/odoo/odoo/blob/0ed63805e93a9e8ec059a496756dfe9b32e66a07/addons/account/models/account_payment.py#L403

```
    @api.depends('partner_id', 'company_id', 'payment_type', 'destination_journal_id', 'is_internal_transfer')
    def _compute_available_partner_bank_ids(self):
        for pay in self:
            if pay.payment_type == 'inbound':
                pay.available_partner_bank_ids = pay.journal_id.bank_account_id
            elif pay.is_internal_transfer:
                pay.available_partner_bank_ids = pay.destination_journal_id.bank_account_id
            else:
                pay.available_partner_bank_ids = pay.partner_id.bank_ids\
                        .filtered(lambda x: x.company_id.id in (False, pay.company_id.id))._origin

    @api.depends('available_partner_bank_ids', 'journal_id')
    def _compute_partner_bank_id(self):
        ''' The default partner_bank_id will be the first available on the partner. '''
        for pay in self:
            pay.partner_bank_id = pay.available_partner_bank_ids[:1]._origin
```

## How to fix it

I fix it overwriting _compute_partner_bank_id function to ensure that this field is not updated if I have a partner_bank_id set previously... this may be not the best way, but I'm not able to find why partner_bank_id is overwrite with company account.

```
class AccountPayment(models.Model):
    _inherit = "account.payment"
    
    @api.depends('available_partner_bank_ids', 'journal_id')
    def _compute_partner_bank_id(self):
        ''' The default partner_bank_id will be the first available on the partner. '''
        for pay in self.filtered(lambda p: not p.partner_bank_id):
            pay.partner_bank_id = pay.available_partner_bank_ids[:1]._origin
```

**Comment by Tecniloop** [comment]
@angelmoya I tested this bug on runboat. It seems is working fine now. Please confirm it. Thanks

**Comment by angelmoya** [comment]
I was not able to reply on rubot, maybe this error is related with other module... But this error I saw in other customers, not only on this project

**Comment by pedrobaeza** [comment]
That's the question, it's not reproducible, and although some of you have expressed to have it, I'm not experimenting it for now, although I don't have a vast instance number on 16 yet.

**Comment by flotho** [comment]
Hi,
I'm facing the same issue on a server.
I didn't succeed in reproducing locally and still investigation
Same repo version

**Comment by flotho** [comment]
small proposal here https://github.com/OCA/bank-payment/pull/1212 to act ASAP in the process before the compute and store method save bad datas.

**Comment by pedrobaeza** [comment]
Solved in #1346
